lib/galaxy/webapps/base/webapp.py

```diff
@@ -190,7 +191,19 @@ def create_mako_template_lookup(self, galaxy_app, name):
             directories=paths, module_directory=galaxy_app.config.template_cache_path, collection_size=500
         )
 
-    def handle_controller_exception(self, e, trans, **kwargs):
+    def handle_controller_exception(self, e, trans, method, **kwargs):
+        if isinstance(e, TypeError):
+            method_signature = inspect.signature(method)
+            required_parameters = {
+                p.name
+                for p in method_signature.parameters.values()
+                if p.name != "trans" and p.default is inspect._empty
+            }
+            missing_required_parameters = required_parameters.difference(kwargs)
+            if missing_required_parameters:
+                e = RequestParameterMissingException(
```

Comment:
Should this be

```suggestion
                raise RequestParameterMissingException(
```

otherwise this is not passed back to the caller.

Reply:
Its a subclass of message exception, which is handled below